app/assets/javascripts/controllers/cloud_volume/cloud_volume_form_controller.js

```diff
@@ -21,19 +21,32 @@ ManageIQ.angular.app.controller('cloudVolumeFormController', ['miqService', 'API
     vm.newRecord = cloudVolumeFormId === 'new';
 
     miqService.sparkleOn();
-    API.get('/api/providers?expand=resources&attributes=id,name,supports_block_storage&filter[]=supports_block_storage=true')
-      .then(getStorageManagers)
-      .catch(miqService.handleFailure);
 
-    if (cloudVolumeFormId !== 'new') {
-      // Fetch cloud volume data before showing the form.
-      API.get('/api/cloud_volumes/' + cloudVolumeFormId + '?attributes=ext_management_system.type,availability_zone.ems_ref,base_snapshot.ems_ref')
-        .then(getCloudVolumeFormData)
-        .catch(miqService.handleFailure);
-    } else {
-      // Initialise the form immediately when creating a new volume.
-      setForm();
+    // Load initial API data depending on what form we're displaying. Do as little requests as possible
+    // to support fine-grained API permissions.
+    var apiPromises = [];
+    switch (cloudVolumeFormOperation) {
+      case 'EDIT':
+        // Fetch relevant StorageManager, just enough to populate the disabled drop-down.
+        apiPromises.push(API.get(`/api/providers/${storageManagerId}?attributes=id,name`)
```

Comment:
Uhh, can't do this here, sorry ;)

Anything under `app/assets` needs to be ES5-only.

Reply:
Thanks for noticing! I've changed as you suggest on both places, no other change, and pushed. Now I'm gone for the weekend :)